src/tests/suite/daos_obj.c

```diff
@@ -4607,6 +4613,15 @@ int_key_setting(void **state)
 
 	rc = daos_obj_close(oh, NULL);
 	assert_rc_equal(rc, 0);
+
+	free(dkey_buf);
+	free(akey_buf);
+}
+
+static void invalid_int_key_setting(void **state)
+{
+	int_key_setting(state, 128);
+	int_key_setting(state, 3);
```

Comment:
I'll let @jolivier23 comment, but i think his proposal was to:

- reject larger ones on client
- reject anything that has odd size (e.g. input must be 1, 2, 4, or 8 bytes)
- I think casting is a little simpler on the server side

Reply:
Thanks, @mchaarawi!  That is outdated though.   Johann and I decided to just reject them if they are less than 8 bytes because there are other places we would need to support this such as in the jump consistent hash for distribution of dkeys and it seemed simpler to just reject the smaller ones.